### PR TITLE
HPC EL9 Transition update

### PR DESCRIPTION
**Pull request by aowen-uwmad**
I'm good with the changes now.

* Added lots of content to the transition page to discuss modules, Spack: https://chtc.github.io/web-preview/preview-hpc-el9-transition-spack/uw-research-computing/hpc-el8-to-el9 
* Updated paths in "Set Up Spack on HPC" guide: https://chtc.github.io/web-preview/preview-hpc-el9-transition-spack/uw-research-computing/hpc-spack-setup
* Restructured "Install Software Using Spack" guide to always start with an interactive job; also added some guidance on checking linking to upstream packages: https://chtc.github.io/web-preview/preview-hpc-el9-transition-spack/uw-research-computing/hpc-spack-install

### CODE REVIEW

File: _uw-research-computing/hpc-el8-to-el9.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+    tag:
         - hpc
 ---
 
 Starting in May 2024, CHTC's high performance computing (HPC) cluster began upgrading
-the Linux distribution and version we use on our servers to **CentOS Stream 9**. This transition is expected to complete in June 2024. 
+the Linux distribution and version we use on our servers to **CentOS Stream 9** (EL9). This transition is expected to complete in June 2024. 
 
 Note that this page only applies to a transition on the HPC cluster. For information 
 on the HTC system transition, see [HPC System Transition to a New Linux Version (CentOS Stream 9)](htc-el8-to-el9.html)
```
Comment (ChristinaLK) [inline _uw-research-computing/hpc-el8-to-el9.md:18]:
```suggestion
on the HTC system transition, see [HPC System Transition to a New Linux Version (CentOS Stream 9)](htc-el8-to-el9-transition.html)
```

File: _uw-research-computing/hpc-el8-to-el9.md [hunk 2]
```diff
@@ -44,20 +44,168 @@ The files in your `/home` and `/scratch` directories will be unchanged.
...
-1. If your jobs no longer run, archive your previous software installation(s) and 
-rebuild your software. Some tips: 
-	* We strongly recommend comping in an interactive job, to confirm your 
-	compiled code will be compatible with all worker nodes in the cluster. 
-	* If you are using a custom spack installation, we recommend archiving your 
-	existing `~/.spack` directory and recreating your spack environment from 
-	scratch as described in [Set Up Spack on HPC](hpc-spack-setup.html)
+3. If your jobs no longer run, archive your previous software installation(s) and 
+   rebuild your software. The [Software Changes](#software-changes) section below has 
+   more specific information.
```
Comment (ChristinaLK) [inline _uw-research-computing/hpc-el8-to-el9.md:53]:
```suggestion
   more specific information about how to do this. 
```

File: _uw-research-computing/hpc-el8-to-el9.md [hunk 3]
```diff
@@ -44,20 +44,168 @@ The files in your `/home` and `/scratch` directories will be unchanged.
...
-please contact the facilitation team at chtc@cs.wisc.edu or [come to office hours](/uw-research-computing/get-help.html)
\ No newline at end of file
+please contact the facilitation team at chtc@cs.wisc.edu or [come to office hours](/uw-research-computing/get-help.html)
+
+## Software Changes
+
+Almost all packages and libraries have been upgraded as part of the operating system transition. 
+Unless your code is fairly simple, you will likely need to recompile it.
+
+**Remember to always compile your code/programs in a (interactive) Slurm job!**
```
Comment (ChristinaLK) [inline _uw-research-computing/hpc-el8-to-el9.md:66]:
```suggestion
**Remember to always compile your code/programs in a (interactive) Slurm job! [How To](hpc-job-submission.html#for-simple-testing-or-compiling)**
```

File: _uw-research-computing/hpc-el8-to-el9.md [hunk 4]
```diff
@@ -44,20 +44,168 @@ The files in your `/home` and `/scratch` directories will be unchanged.
...
+> Different versions of module packages, or packages that are "dropped" or "deprecated" may be manually installed by the user using [Spack](#spack).
+
+### Spack
+
+Spack is a package manager platform that allows users to install software without admin privileges.
+CHTC also uses Spack to install the software underlying the system-wide modules discussed above.
+
+> If you have not used Spack before, you can skip this section and go directly to the [Set Up Spack on HPC](hpc-spack-setup.html) guide.
+
+Here is the general process for setting up your software on the upgraded EL9 system:
```
Comment (ChristinaLK) [inline _uw-research-computing/hpc-el8-to-el9.md:123]:
```suggestion
Here is the general process for setting up your software on the upgraded EL9 system; detailed instructions are provided after the general process: 
```

File: _uw-research-computing/hpc-el8-to-el9.md [hunk 5]
```diff
@@ -44,20 +44,168 @@ The files in your `/home` and `/scratch` directories will be unchanged.
...
+The login server uses different hardware than the execute servers, and the mismatch leads to Spack using the wrong settings for installing packages.
+Of course, as before, you should only install packages while in interactive Slurm job.
+
+Behind the scenes, we've made a few changes to the configuration that will hopefully make the package installation much smoother.
+
+#### 5. Update your workflow
+
+Finally, remember to update your workflow to use the new Spack environments and the packages installed therein.
+
+* If you explicitly provid paths to packages installed using Spack, be sure to update those paths in your compiler configuration or in your job submission script.
```
Comment (ChristinaLK) [inline _uw-research-computing/hpc-el8-to-el9.md:207]:
```suggestion
* If you explicitly provide paths to packages installed using Spack, be sure to update those paths in your compiler configuration or in your job submission script.
```

**Review by ChristinaLK — APPROVED** [review]
Looks good! A few tweaks in comments.

File: _uw-research-computing/hpc-el8-to-el9.md [hunk 6]
```diff
@@ -1,31 +1,31 @@
...
         - hpc
 ---
 
 Starting in May 2024, CHTC's high performance computing (HPC) cluster began upgrading
-the Linux distribution and version we use on our servers to **CentOS Stream 9**. This transition is expected to complete in June 2024. 
+the Linux distribution and version we use on our servers to **CentOS Stream 9** (EL9). This transition is expected to complete in June 2024. 
 
 Note that this page only applies to a transition on the HPC cluster. For information 
-on the HTC system transition, see [HPC System Transition to a New Linux Version (CentOS Stream 9)](htc-el8-to-el9.html)
+on the HTC system transition, see [HPC System Transition to a New Linux Version (CentOS Stream 9)](htc-el8-to-el9-transition.html)
```
Comment (aowen-uwmad) [inline _uw-research-computing/hpc-el8-to-el9.md:18]:
```suggestion
on the HTC system transition, see [HTC System Transition to a New Linux Version (CentOS Stream 9)](htc-el8-to-el9-transition.html)
```

**Pull request merged**